source/upgrade/upgrade-mattermost-kubernetes-ha.rst

```diff
@@ -0,0 +1,232 @@
+
+1. **Cluster health**: Ensure all nodes are in a ready state and all pods are running without errors. This step is crucial to avoid issues during the upgrade process. Use the following commands to check cluster health:
+
+  .. code-block:: bash
+
+    kubectl get nodes
+    kubectl get pods --all-namespaces
+
+2. **Helm setup**: Verify that Helm is installed and configured correctly in your Kubernetes cluster. Ensure you have the latest version of the Mattermost Operator Helm chart. Use the following commands to check Helm setup:
+
+  .. code-block:: bash
+
+    helm version
+    helm repo update
+    helm search repo mattermost/mattermost-operator
+
+3. Confirm Mattermost Operator version. Make sure the image version matches the latest supported release from the Helm repository. Use the following command to check the current image version of the Mattermost Operator:
+
+  .. code-block:: bash
+
+    kubectl get deployment mattermost-operator -n mattermost -o=jsonpath='{.spec.template.spec.containers[0].image}'
+
+4. **Verify available resources**: Ensure your Kubernetes cluster has sufficient CPU and memory resources to handle the upgrade process. Use the following commands to check resource availability:
+
+  .. code-block:: bash
+
+    kubectl top nodes
+    kubectl describe node | grep Allocatable
+
+5. Back up database and file storage.
+
+    Use ``pg_dump`` or volume snapshots to create a full backup to a secure, external location (e.g., S3 or NFS). Validate the backup can be restored. See the :doc:`Backup and Disaster Recovery </deploy/backup-disaster-recovery>` documentation for details.
+
+6. Ensure configuration consistency: Validate that ``values.yaml``, secrets, and other configuration files are version-controlled and consistent across all clusters (required for `Active/Active deployments <#active/active-deployments>`__). Use tools like GitOps or configuration management systems to ensure all clusters have the same configuration.
+
+7. We strongly recommend performing a dry run by testing the upgrade in a staging environment that mirrors production to catch misconfigurations early.
+
+Step 2: Perform the upgrade
+----------------------------
+
+When you use the Mattermost Operator, the Operator handles the upgrade process automatically, ensuring that pods are updated incrementally and that traffic is routed correctly throughout the upgrade.
+
+The Mattermost Operator completes the following actions:
+
+- Performs pre-upgrade compatibility checks. 
+- Halts the upgrade if issues are detected.
+- Rolls out upgrades incrementally across pods.
+- Maintains live traffic routing during pod transitions.
+
+Failed upgrades don't modify the live cluster and are automatically rolled back, ensuring the cluster remains stable. Health monitoring ensures that only healthy pods are replaced, and new pods are brought online only after passing health checks. New pods are deployed with the updated version, while old pods are gracefully terminated.
```

Comment:
So the upgrade process is not directly rolled back. The database could be in a partially upgraded state. We probably want to change this line to indicate that the admin can either perform a rollback manually (not sure if we have docs on that elsewhere) or they can review the server logs to see what the error is and address it.